TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/AuthScreen/AuthScreenViewController.m

```diff
@@ -0,0 +1,339 @@
+//
+//  AuthScreenViewController.m
+//  TestAssignmentDoItSoftware
+//
+//  Created by Vladimir Samoylenko on 2/15/19.
+//  Copyright © 2019 TheRedQueen. All rights reserved.
+//
+
+#import "AuthScreenViewController.h"
+
+@interface AuthScreenViewController ()
+
+@end
+
+@implementation AuthScreenViewController
+
+
+@synthesize tfLogin;
+@synthesize tfPassword;
+@synthesize btnAuth;
+@synthesize swMode;
+
+
+#pragma mark - screen related
+
+- (void)viewDidLoad {
+
+    [super viewDidLoad];
+
+}
+
+
+-(void)viewWillAppear:(BOOL)animated{
+    
+    [super viewWillAppear:animated];
+    
+    UIColor *grayColor = [[UIColor alloc] initWithRed:130.0/255.0 green:130.0/255.0 blue:130.0/255.0 alpha:130.0/255.0];
+
+    tfLogin.layer.cornerRadius = 2.0;
+    tfLogin.layer.borderWidth = 1.0;
+    tfLogin.layer.borderColor = grayColor.CGColor;
+
+    tfPassword.layer.cornerRadius = 2.0;
+    tfPassword.layer.borderWidth = 1.0;
+    tfPassword.layer.borderColor = grayColor.CGColor;
+
+    btnAuth.layer.cornerRadius = 2.0;
+    btnAuth.layer.borderWidth = 1.0;
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"registrationfailed" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(authFailed:) name:@"registrationfailed" object:nil];
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"registrationdone" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(authCompleted:) name:@"registrationdone" object:nil];
+
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authfailed" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(authFailed:) name:@"authfailed" object:nil];
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authdone" object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(authCompleted:) name:@"authdone" object:nil];
+
+}
+
+
+-(void)viewWillDisappear:(BOOL)animated{
+    
+    [super viewWillDisappear:animated];
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"registerfailed" object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"registrationdone" object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authfailed" object:nil];
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:@"authdone" object:nil];
+
+}
+
+
+-(void)viewDidAppear:(BOOL)animated{
+    
+    [super viewDidAppear:animated];
+    
+}
+
+
+#pragma mark - app flow actions
+
+- (IBAction)btnAuthTapped:(id)sender{
+
+    NSLog(@"btnAuth");
+
+    if ([self checkProvidedValues] == true){
+        
+        [self showActivity];
+        
+        TASingleton *singleton = [[TASingleton alloc] init];
```

Comment:
Either this should be a call to `sharedInstance` or the singleton use is unclear.